text/0000-sensitive-data-redaction.md

```diff
@@ -0,0 +1,276 @@
+# Sensitive Data Redaction
+
+This is a proposal for adding treatment of sensitive data to the OpenTelemetry (OTel) Specification and semantic conventions.
+
+## Motivation
+
+When collecting data from an application, there is always the possibility, that the data contains information that shouldn’t be collected, because it is either leaking (parts of) credentials (passwords, tokens, usernames, credit card information) or can be used to uniquely identify a person (name, IP, email, credit card information) which may be protected through certain regulations. By adding OTel to a library or instrumentation an end-user of OTel is facing exactly this challenge: values of an [Attribute](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute) may carry such sensitive data.
```

Comment:
In additional to "values of an Attribute", the name of the span, name of the event and any other strings that come from the user could share the same problem.

Consider something like "any string value that comes from the user could carry sensitive data".

In theory non-string types may also have the issue (e.g. if folks embed identifiable information in a GUID), probably fine to leave it as doing such thing is already very anti-pattern.

Reply:
Thanks, I can update it accordingly. I agree that non-string values _could_ have a similar issue but it is relatively unlikely

Reply:
I kept it as generic as possible: `emitted telemetry may carry such sensitive data.`